### PR TITLE
Add 'AzureBastionSubnet' exclusion for UDR on Subnets Policy

### PR DESCRIPTION
**Pull request by jtracey93**
**This PR fixes:**
- #604 

**Summary of changes:**
- Added exclusion of `AzureBastionSubnet` in `Deny-Subnet-Without-UDR` policy
  - In all 4 reference implementations
- Update `ESLZ-Policies.md` with a note for this policy detailing this

### CODE REVIEW

**Comment by jtracey93** [comment]
Testing Evidence:
image

**Comment by krnese** [comment]
Thanks. Should we cater for the other subnets as well and add the in clause instead of equals towards a single subnet?

**Comment by jtracey93** [comment]
No problem. And yes I agree we could make it an array to be more inclusive and useful 👍

Also myself and @victorar have been discussing how we ensure the exclusions are not misused by landing zones, which I think I have a fix for (with another policy). 

Let's sync next week and discuss as a group 👍

**Review by uday31in — APPROVED** [review]

File: docs/reference/adventureworks/armTemplates/auxiliary/policies.json [hunk 1]
```diff
@@ -10677,6 +10677,12 @@
                     "field": "type",
                     "equals": "Microsoft.Network/virtualNetworks/subnets"
                   },
+                  {
+                    "not": {
+                      "field": "name",
+                      "equals": "AzureBastionSubnet"
```
Comment (krnese) [inline docs/reference/adventureworks/armTemplates/auxiliary/policies.json:10683]: should we parameterize the value (AzureBastionSubnet) and change this to be "in"?

Comment (jtracey93) [reply docs/reference/adventureworks/armTemplates/auxiliary/policies.json:10683]: Lets discuss on our call later in the week. But I agree and think we should likely make this an Array so its more flexible for other subnets

Comment (jtracey93) [reply docs/reference/adventureworks/armTemplates/auxiliary/policies.json]: This has now been done in the latest commits

**Review by victorar — DISMISSED** [review]

File: docs/ESLZ-Policies.md [hunk 2]
```diff
@@ -104,7 +104,7 @@ We work with - and learn from our customers and partners to ensure that we evolv
...
 | Deny vNet peering | Denies creation of vnet peering | deny, audit, disabled | 1.0.0 | Custom policy | Yes, recommended | Sandboxes Management Group | Network |
 | Deny vNet peering cross subscription | Denies the creation of vnet peering outside of the same subscription | deny, audit, disabled | 1.0.0 | Custom policy | No | Sandboxes Management Group | Network |
-| Subnets should have a User Defined Route | Denies the creation of a subnet without having a User-Defined Route | deny, audit, disabled | 1.0.0 | Custom policy | No |  | Network |
+| Subnets should have a User Defined Route | Denies the creation of a subnet without having a User-Defined Route, except the `AzureBastionSubnet` as this is not supported as documented [here](https://docs.microsoft.com/azure/bastion/bastion-overview#udr). The `AzureBastionSubnet` is only allowed to contain the the Azure Bastion service, this is enforced by the Azure platform itself. | deny, audit, disabled | 1.1.0 | Custom policy | No | Landing Zones Management Group | Network |
```
Comment (victorar) [inline docs/ESLZ-Policies.md:107]: Pls remove the duplicated word:

"the the Azure Bastion"

**Review by victorar — CHANGES_REQUESTED** [review]

File: docs/ESLZ-Policies.md (hunk 2 above)
Comment (jtracey93) [reply docs/ESLZ-Policies.md]: Good spot, resolved 👍

**Review by victorar — APPROVED** [review]

**Review by krnese — APPROVED** [review]

**Pull request merged**